adjusted test_transformations.py::test_filter fails randomly

Merging https://github.com/bids-standard/pybids/pull/413 did bring back that failure I saw but which went away after rebasing on master, but not for long - apparently there is some randomless somewhere, and specific to python 3.5, see e.g. https://travis-ci.org/bids-standard/pybids/builds/509649783  build

```
collection = <bids.variables.kollekshuns.BIDSRunVariableCollection object at 0x7f66900471d0>
    def test_filter(collection):
        orig = collection['parametric gain'].clone()
        q = "parametric gain > 0"
        transform.Filter(collection, 'parametric gain', query=q)
        assert len(orig.values) == 2 * len(collection['parametric gain'].values)
        assert np.all(collection['parametric gain'].values > 0)
    
        orig = collection['RT'].clone()
        q = 'parametric gain > 0.1'
        transform.Filter(collection, 'RT', query=q, by='parametric gain')
        assert len(orig.values) != len(collection['RT'].values)
        # There is some bizarro thing going on where, on travis, the result is
        # randomly either 1536 (when all subjects are used) or 3909 when running
        # on Python 3 (on linux or mac).
        # Never happens locally, and I've had no luck tracking down the problem.
        # Best guess is it reflects either some non-deterministic ordering of
        # variables somewhere, or some weird precision issues when resampling to
        # dense. Needs to be tracked down and fixed.
>       assert len(collection['RT'].values) in [96 * len(SUBJECTS), 3909]
E       assert 499 in [192, 3909]
E        +  where 499 = len(1      1.793\n2      1.637\n3      1.316\n4      1.670\n5      1.232\n6      1.502\n7      1.925\n8      1.061\n9      1.079\n1...289\n505    2.976\n506    1.257\n507    1.199\n508    1.432\n509    1.440\n510    2.071\nName: RT, Length: 499, dtype: float64)
E        +    where 1      1.793\n2      1.637\n3      1.316\n4      1.670\n5      1.232\n6      1.502\n7      1.925\n8      1.061\n9      1.079\n1...289\n505    2.976\n506    1.257\n507    1.199\n508    1.432\n509    1.440\n510    2.071\nName: RT, Length: 499, dtype: float64 = <bids.variables.variables.SparseRunVariable object at 0x7f66903d1a20>.values
bids/analysis/tests/test_transformations.py:382: AssertionError
```
I have reproduced it locally in venv with 3.5,  fails some times... 50/50. looking into it to get more details

--- a/bids/analysis/tests/test_transformations.py
+++ b/bids/analysis/tests/test_transformations.py
@@ -372,14 +372,7 @@ def test_filter(collection):
     q = 'parametric gain > 0.1'
     transform.Filter(collection, 'RT', query=q, by='parametric gain')
     assert len(orig.values) != len(collection['RT'].values)
-    # There is some bizarro thing going on where, on travis, the result is
-    # randomly either 1536 (when all subjects are used) or 3909 when running
-    # on Python 3 (on linux or mac).
-    # Never happens locally, and I've had no luck tracking down the problem.
-    # Best guess is it reflects either some non-deterministic ordering of
-    # variables somewhere, or some weird precision issues when resampling to
-    # dense. Needs to be tracked down and fixed.
-    assert len(collection['RT'].values) in [96 * len(SUBJECTS), 3909]
+    assert len(collection['RT'].values) == 96 * len(SUBJECTS)
 
 
 def test_replace(collection):

--- a/bids/analysis/transformations/munge.py
+++ b/bids/analysis/transformations/munge.py
@@ -5,11 +5,16 @@ other formats or shapes.
 
 import numpy as np
 import pandas as pd
+# Might not be needed since python 3.7
+# See https://github.com/bids-standard/pybids/issues/423 for more info
+from collections import OrderedDict as odict
+
 from bids.utils import listify
 from .base import Transformation
 from patsy import dmatrix
 import re
 from bids.variables import DenseRunVariable, SimpleVariable
+
 
 class Assign(Transformation):
     ''' Assign one variable's amplitude, duration, or onset attribute to
@@ -160,6 +165,7 @@ class Factor(Transformation):
 
 
 class Filter(Transformation):
+    '''Filter (remove) the values given a query to satisfy'''
 
     _groupable = False
     _input_type = 'variable'
@@ -174,13 +180,15 @@ class Filter(Transformation):
 
         names = [var.name] + listify(by)
 
+        # assure ordered dict so we have consistent (if not correct) operation,
+        # because later we ask for name_map.values
         # pandas .query can't handle non-identifiers in variable names, so we
         # need to replace them in both the variable names and the query string.
-        name_map = {n: re.sub('[^a-zA-Z0-9_]+', '_', n) for n in names}
+        name_map = odict((n, re.sub('[^a-zA-Z0-9_]+', '_', n)) for n in names)
         for k, v in name_map.items():
             query = query.replace(k, v)
 
-        data = pd.concat([self.collection[c].values for c in names],
+        data = pd.concat([self.collection[n].values for n in names],
                          axis=1, sort=True)
         # Make sure we can use integer index
         data = data.reset_index(drop=True)

